README.md

```diff
@@ -34,14 +43,143 @@ then `func` can discover it and expose it to a user in a similar way the static
   project creation (e.g. group and artifact ids for maven project).
 * The `executable` behaviour may be customized by environment variables.
 
+The advantage of the `Contract A` is simpler implementation,
+but it would be usable mostly just for `func`.
+
+## Executable Contract B
+
+* The `executable` shall support three sub-commands: `runtimes`, `templates` and `create`.
+* The `runtimes` sub-command shall take no parameters.</br>
+  The sub-command shall print a JSON array of string to standard output containing a list of supported runtimes.
+* The `templates` sub-command shall take exactly one parameter: runtime name. </br>
+  The sub-command shall print a JSON Object where keys are names of templates and values are specification
+  of questions/parameters needed by a template.</br>
+  Example of such a JSON: 
```

Comment:
Is this an example of a single template?
How does this look if we had a repository of templates? (like func-tastic)